Library/Homebrew/cmd/update-report.rb

```diff
@@ -313,18 +333,22 @@ def initialize(var_name)
     end
   end
 
-  attr_reader :tap, :initial_revision, :current_revision
-
-  def initialize(tap)
+  def initialize(tap, api_names_txt: nil, api_names_before_txt: nil)
     @tap = tap
+    @api_names_txt = api_names_txt
+    @api_names_before_txt = api_names_before_txt
 
-    initial_revision_var = "HOMEBREW_UPDATE_BEFORE#{tap.repo_var}"
-    @initial_revision = ENV[initial_revision_var].to_s
-    raise ReporterRevisionUnsetError, initial_revision_var if @initial_revision.empty?
+    if api_names_txt.present? && api_names_before_txt.present?
```

Comment:
Linter is annoyed here 😁 

Can you do 
```
if !api_names_txt.present? && !api_names_before_txt.present?
```
in Ruby and then remove this blank condition? (not a Rubyist for years and years)

Reply:
@ocean This is WIP and not expected to work at all yet.